WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/RestoreStates.kt

```diff
@@ -0,0 +1,106 @@
+package org.wordpress.android.ui.jetpack.restore
+
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.model.activity.ActivityLogModel
+import org.wordpress.android.ui.jetpack.common.JetpackListItemState
+import org.wordpress.android.ui.jetpack.restore.StateType.DETAILS
+import org.wordpress.android.ui.jetpack.restore.StateType.WARNING
+import org.wordpress.android.ui.jetpack.restore.StateType.PROGRESS
+import org.wordpress.android.ui.jetpack.restore.StateType.COMPLETE
+import org.wordpress.android.ui.jetpack.restore.StateType.ERROR
+
+abstract class RestoreUiState(open val type: StateType) {
+    abstract val items: List<JetpackListItemState>
+    abstract val toolbarState: ToolbarState
+
+    data class ErrorState(
+        val errorType: RestoreErrorTypes,
+        override val items: List<JetpackListItemState>,
+        override val toolbarState: ToolbarState
+    ) : RestoreUiState(ERROR)
+
+    sealed class ContentState(override val type: StateType) : RestoreUiState(type) {
+        data class DetailsState(
+            val activityLogModel: ActivityLogModel,
+            override val items: List<JetpackListItemState>,
+            override val toolbarState: ToolbarState,
+            override val type: StateType
+        ) : ContentState(DETAILS)
+
+        data class WarningState(
+            override val items: List<JetpackListItemState>,
+            override val toolbarState: ToolbarState,
+            override val type: StateType
+        ) : ContentState(WARNING)
+
+        data class ProgressState(
+            override val items: List<JetpackListItemState>,
+            override val toolbarState: ToolbarState,
+            override val type: StateType
+        ) : ContentState(PROGRESS)
+
+        data class CompleteState(
+            override val items: List<JetpackListItemState>,
+            override val toolbarState: ToolbarState,
+            override val type: StateType
+        ) : ContentState(COMPLETE)
+    }
+}
+
+enum class StateType(val id: Int) {
+    DETAILS(0),
+    WARNING(1),
+    PROGRESS(2),
+    COMPLETE(3),
+    ERROR(4)
+}
+
+sealed class ToolbarState {
+    abstract val title: Int
+    @DrawableRes val icon: Int = R.drawable.ic_close_24px
+
+    data class DetailsToolbarState(
+        @StringRes override val title: Int = R.string.restore_details_page_title
```

Comment:
⚠️ We might want to turn this into a regular field (move it out of the constructor), since we always set this value to `R.string.restore_details_page_title`. Wdyt?

```
object DetailsToolbarState : ToolbarState() {
        @StringRes override val title: Int = R.string.restore_details_page_title
    }
```

Reply:
Good idea